src/cdk/testing/tests/protractor.e2e.spec.ts

```diff
@@ -6,10 +6,19 @@ import {
 } from '@angular/cdk/testing';
 import {expectAsyncError} from '@angular/cdk/testing/private';
 import {ProtractorHarnessEnvironment} from '@angular/cdk/testing/protractor';
-import {browser} from 'protractor';
+import {browser, by, element as protractorElement, ElementFinder} from 'protractor';
 import {MainComponentHarness} from './harnesses/main-component-harness';
 import {SubComponentHarness, SubComponentSpecialHarness} from './harnesses/sub-component-harness';
 
+// Kagekiri is available globally in the browser. We declare it here so we can use it in the
+// browser-side script passed to `by.js`.
+declare const kagekiri: {
+  querySelectorAll: (selector: string, root: Element) => NodeListOf<Element>;
+};
```

Comment:
Could this declared const use the types from the `kagekiri.d.ts` file? (to avoid duplication). I think you could use a type-only import to the module.

Reply:
I don't think type only imports are available in TS 3.6, no?

Reply:
It looks like that's only available in TS 3.8, I left a comment to clean it up later